frontend/src/services/ErrorHandler.ts

```diff
@@ -0,0 +1,36 @@
+import {Dispatch} from 'redux'
+import { toast } from 'react-toastify'
+
+import {errorMessage } from '../constants/errorAndSuccessMessages'
+import {userPreferencesRequestFailed} from '../actions/prefAndProfileActions'
+
+export interface IError {
+        field: string
+        message: string
+        statusCode?: number // код статуса ошибки например 401 или 500
+        errorType?: string // тип ошибки например ошибка формы
+}
+
+export interface IErrorHandlerReturn {
+    errorText: string | string[]
+}
+
+export const errorHandler = (error: IError, dispatch: Dispatch): void => {
+    try {
+        // here we check that error object is in JSON format
+        const destructuredMessage = JSON.parse(error.message)
+        if (destructuredMessage) {
+            const [messageArrayFromDestructuredError] = destructuredMessage.errors
+            const errorText = (messageArrayFromDestructuredError.message).toString()
+            console.log(messageArrayFromDestructuredError.message)
+            dispatch(userPreferencesRequestFailed(errorText))
```

Comment:
`userPreferencesRequestFailed` does it mean that error handler will handle only user preferences error?

Reply:
I added field ErrorType for 'Authorization' errors, as they are not placed into State and can be just shown as toast notifications. Other but authorization is handled by `userPreferenceRequestFailed`.